### DISCUSSION TITLE
[API] get-user endpoint with built-in user

### DISCUSSION THREAD

**Discussion by m-i-k-e-e** [casdoor/casdoor#2735] [Q&A] [answered]
Hi, 

I'm trying to get  a user detail using the API.
If I use a user/password authentication, the get-user endpoint work fine.

If I use the client_crendentials of the built-in user, I got the following error
```
{
  "status": "error",
  "msg": "Session outdated, please login again",
  "sub": "",
  "name": "",
  "data": null,
  "data2": null
}
```

On the other side, the get-users endoint works fine for both (built-in & user)

Does anybody have an idea ?

Thanks

[JMeter sample](https://github.com/casdoor/casdoor/files/14384361/HTTP.Request.zip)

**Comment by hsluoyz** [comment, marked as answer]
@m-i-k-e-e see docs: https://casdoor.org/docs/basic/public-api#how-to-authenticate-with-casdoor-public-api

**Reply by m-i-k-e-e** [reply]
I already do that. It works fine with de get-users endpoint. It's only the get-user that gives me issues.
it feels like the get-user endpoint doesn't work with the built-in application credentials

**Reply by hsluoyz** [reply]
@m-i-k-e-e can you give a URL example for demo site: https://door.casdoor.com/ ?

**Reply by m-i-k-e-e** [reply]
I tested with this site, and it works.
I also use the api to create the organization, application and user.
I tried to create all of them thru casdoor web pages and it works.
I probably forget to initialize a field which causes a strange behaviour.
I'll try to figure out and post an update to this thread.

Thanks for your help

**Comment by m-i-k-e-e** [comment]
@hsluoyz 
the is_profile_public field in the organization table was set to false which causes the error.
The error message is a little bit weird
I set it to true and the get-user endpoint works perfectly

**Reply by hsluoyz** [reply]
I don't have such issue

![image](https://github.com/casdoor/casdoor/assets/3787410/11e7761c-0072-4582-9858-94a8a9d0882e)

**Reply by m-i-k-e-e** [reply]
1. set the is_profile_public to false for the organization (ex : 123) 
2. get a token using the built-in client credentials
3. using this token as a bearer token, reach the get-user endpoint with id = 123/username
